src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/TickerTask.java

```diff
@@ -154,13 +147,14 @@ private void tickLocation(@Nonnull Set<BlockTicker> tickers, @Nonnull Location l
         Config data = BlockStorage.getLocationInfo(l);
         SlimefunItem item = SlimefunItem.getByID(data.getString("id"));
 
-        if (item != null && item.getBlockTicker() != null) {
+        // Fixes #2954
+        if (item != null && !item.isDisabledIn(l.getWorld()) && item.getBlockTicker() != null) {
```

Comment:
Checking this on every tick is extremely bad and can weigh everything down.
We should rather not even mark this as a ticker if it is disabled anyway, so it won't even show up in this `Set`.

Reply:
Tested with 101x101x101 Rainbow Glass:

Rainbow Glass is enabled:
![image](https://user-images.githubusercontent.com/34732414/115180493-e2605200-a0ff-11eb-9f5b-cd21ee1925b4.png)

Rainbow Glass is disabled:
![image](https://user-images.githubusercontent.com/34732414/115180572-189dd180-a100-11eb-9904-8ac0656340a7.png)

idk if this is relevant or not

Reply:
The impact might be small, but it is avoidable and any optimizations are good, small and large.

Reply:
Will do this but also worth noting that this wont cover different worlds. Can't think of a better way to check it than every tick in this case.

Reply:
Tickers are handled per-world anyway.